### PR TITLE
.Net: Remove Kernel.Builder

### PR DESCRIPTION
**Pull request by feiyun0112**
Resolves: #3169 
Use `new KernelBuilder()` instead of `Kernel.Builder`

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/Example42_KernelBuilder.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 // ==========================================================================================================
 // The easier way to instantiate the Semantic Kernel is to use KernelBuilder.
-// You can access the builder using either Kernel.Builder or KernelBuilder.
+// You can access the builder using either new KernelBuilder() or KernelBuilder.
```
Comment (dmytrostruk) [inline dotnet/samples/KernelSyntaxExamples/Example42_KernelBuilder.cs:5]: I think now there will be only one way how to access `KernelBuilder` - it's `new KernelBuilder()`. I think this comment can be removed.
```suggestion
```

File: dotnet/src/Plugins/Plugins.Core/ConversationSummaryPlugin.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.SemanticKernel.Plugins.Core;
 /// </summary>
 /// <example>
 /// <code>
-/// var kernel Kernel.Builder.Build();
+/// var kernel new KernelBuilder().Build();
```
Comment (dmytrostruk) [inline dotnet/src/Plugins/Plugins.Core/ConversationSummaryPlugin.cs:17]:
```suggestion
/// var kernel = new KernelBuilder().Build();
```

File: dotnet/src/SemanticKernel.Core/Kernel.cs [hunk 3]
```diff
@@ -41,11 +41,6 @@ public sealed class Kernel : IKernel, IDisposable
     /// <inheritdoc/>
     public IPromptTemplateEngine PromptTemplateEngine { get; }
 
-    /// <summary>
-    /// Return a new instance of the kernel builder, used to build and configure kernel instances.
-    /// </summary>
-    public static KernelBuilder Builder => new();
```
Comment (dmytrostruk) [inline dotnet/src/SemanticKernel.Core/Kernel.cs:47]: This is definitely breaking change. I would propose to keep all changes in this PR as it is but mark this field as `Obsolete` for some period of time, so users will be able to migrate to another syntax smoothly. 
```csharp
/// <summary>
/// Return a new instance of the kernel builder, used to build and configure kernel instances.
/// </summary>
[Obsolete("This field will be removed in a future release. Initialize KernelBuilder through constructor instead (new KernelBuilder()).")]
public static KernelBuilder Builder => new();
```

**Review by dmytrostruk — COMMENTED** [review]
@feiyun0112 Thanks a lot! 
I left a couple of comments to resolve.

File: samples/dotnet/OpenApiPluginsExample/Program.cs [hunk 4]
```diff
@@ -51,7 +51,7 @@ private static async Task Main(string[] args)
         AIServiceOptions aiOptions = configuration.GetRequiredSection(AIServiceOptions.PropertyName).Get<AIServiceOptions>()
             ?? throw new InvalidOperationException($"Missing configuration for {AIServiceOptions.PropertyName}.");
 
-        KernelBuilder kb = Kernel.Builder
+        KernelBuilder kb = new KernelBuilder()
```
Comment (RogerBarreto) [inline samples/dotnet/OpenApiPluginsExample/Program.cs:54]:
```suggestion
        KernelBuilder builder = new KernelBuilder()
```

**Review by RogerBarreto — APPROVED** [review]
Fix conflicts, LGTM

**Review by dmytrostruk — APPROVED** [review]

**Pull request merged**